Sources/BSON/Date+BSONValue.swift

```diff
@@ -33,7 +33,12 @@ extension Date: BSONValue {
             self = Date(msSinceEpoch: int)
         case let .string(s):
             // relaxed extended JSON
-            guard let date = ExtendedJSONDecoder.extJSONDateFormatter.date(from: s) else {
+            // If fractional seconds are omitted in the input (length is 20 instead of 23),
+            // formatter should only account for seconds, otherwise formatter should take milliseconds into account          // Otherwise, fractional seconds SHOULD be omitted if zero.
+            let formatter = s.count == 20
```

Comment:
This hard coded length requirement is probably not ideal, any alternate ways to go about distinguishing between seconds and milliseconds formatting?

Reply:
ok done